misc/scripts/install-local.sh

```diff
@@ -160,29 +160,53 @@ function makeVirtualDeb {
 		printf "Version: 0%s-1\n" "$version" | sudo tee -a "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
 	fi
 
-	if [[ -n $depends ]]; then
-		printf "Depends: %s\n" "${depends//' '/' , '}"| sudo tee -a "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
-	fi
-
-	if [[ -n $optdepends ]]; then
-		fancy_message info "$name has optional dependencies that can enhance its functionalities"
-		echo "Optional dependencies:"
-		printf '    %s\n' "${optdepends[@]}"
-		ask "Do you want to install them" Y
-		if [[ $answer -eq 1 ]]; then
-			optinstall='--install-suggests'
-		fi
-
-		printf "Suggests:" |sudo tee -a "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
+	if [[ ${#optdepends[@]} -ne 0 ]]; then
 		for i in "${optdepends[@]}"; do
 			if ! grep -q ':' <<< "${i}"; then
 				fancy_message error "${i} does not have a description"
 				cleanup
 				return 1
 			fi
 		done
-		printf " %s\n" "${optdepends[@]}" | awk -F': ' '{print $1}' | tr '\n' ',' | head -c -1 | sudo tee -a "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
-		printf "\n" | sudo tee -a "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
+		
+		
+		optdeps=()
+		for optdep in "${optdepends[@]}"; do
+			opt=${optdep%%: *}
+			# Add to the dependency list if already installed so it doesn't get autoremoved on upgrade
+			# Add to the optdeps list if not to display the question
+			if ! dpkg-query -W -f='${Status}' "${opt}" 2> /dev/null | grep "^install ok installed" > /dev/null 2>&1; then
+				optdeps+=("${optdep}")
+			else
+				depends+=" ${opt}"
+			fi
+		done
+
+		if [[ -n $optdeps ]]; then
+			fancy_message info "$name has optional dependencies that can enhance its functionalities"
+			echo "Optional dependencies:"
+			printf '    %s\n' "${optdeps[@]}"
+			ask "Do you want to install them" Y
+			if [[ $answer -eq 1 ]]; then
+				for optdep in "${optdeps[@]}"; do
+					depends+=" ${optdep%%: *}"
+				done
+				if pacstall -L | grep "$name" > /dev/null 2>&1; then
```

Comment:
This grep should match the entire line, not just a match, because a package could also have a `-git`, which would also be matched.

Reply:
True, there is even the fixed version somewhere in the code